applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_custom_scipy_solver.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+# Import base class file
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_solver import MechanicalSolver
+
+from KratosMultiphysics import python_linear_solver_factory as linear_solver_factory
+
+# Import scipy modules
+import KratosMultiphysics.scipy_conversion_tools
+
+def CreateSolver(main_model_part, custom_settings):
+    return CustomScipySolver(main_model_part, custom_settings)
+
+class CustomScipySolver(MechanicalSolver):
```

Comment:
```suggestion
class MatrixBuilder(MechanicalSolver):
```

Reply:
:+1: or ScipyMatrixBuilder

Reply:
Ouch i tend to disagree on this. this is a complete solver, which can be integrated together with the standard analysis stage...

Reply:
Now, with the implementation on SolveSolutionStep it would be something like "ScipyEigenvalueSolver"

Reply:
true, nevertheless what is implemented there is just an example of what can be done... @manuelmessmer can u add a comment to tell that eigenvalue solution is only an example and that one should derive his own implementation from this "base solver"?

Reply:
Yes will do. I am also working on a test.
So I will keep the name "CustomScipySolver"?